gpytorch/likelihoods/gaussian_likelihood.py

```diff
@@ -1,42 +1,56 @@
 from __future__ import absolute_import, division, print_function, unicode_literals
 
 import math
-import torch
-from ..distributions import MultivariateNormal
-from ..functions import add_diag
-from ..likelihoods import Likelihood
-from .. import settings
+import warnings
+import logging
 
+from .. import settings
+from ..distributions import MultivariateNormal
+from ..lazy import AddedDiagLazyTensor, DiagLazyTensor
+from .likelihood import Likelihood
+from .noise_models import HomoskedasticNoise
 
-class GaussianLikelihood(Likelihood):
-    r"""
-    """
-
-    def __init__(self, log_noise_prior=None, batch_size=1):
-        super(GaussianLikelihood, self).__init__()
-        self.register_parameter(
-            name="log_noise", parameter=torch.nn.Parameter(torch.zeros(batch_size, 1)), prior=log_noise_prior
-        )
+DEPRECATION_WARNING = "'GaussianLikelihood' was renamed to 'HomoskedasticGaussianLikelihood'"
 
-    @property
-    def noise(self):
-        return self.log_noise.exp()
 
-    def forward(self, input):
+class GaussianLikelihood(Likelihood):
```

Comment:
I'd prefer if `GaussianLikelihood` became `_GaussianLikelihoodBase` and `HomoskedasticGaussianLikelihood` became `GaussianLikelihood` (with `HeteroskedasticGaussianLikelihood` remaining the same). I think homoskedastic is the common case, and this avoids making this a breaking change

Reply:
that's fine by me